h/views/api_auth.py

```diff
@@ -85,6 +112,22 @@ def _authorized_response(self):
             client_id = self.request.params.get('client_id')
             raise RuntimeError('created authorisation code for client "{}" but got no redirect location'.format(client_id))
 
+    def _render_web_message_response(self, redirect_uri):
+        location = urlparse.urlparse(redirect_uri)
+        params = urlparse.parse_qs(location.query)
+        origin = '{l.scheme}://{l.netloc}'.format(l=location)
+
+        state = None
+        states = params.get('state', [])
+        if states:
+            state = states[0]
+
+        return {
+            'code': params.get('code', [])[0],
```

Comment:
Indeed. `state` is recommended but optional according to https://tools.ietf.org/html/rfc6749#section-4.1.1